packages-ts/gauntlet-terra-contracts/src/commands/abstract/batchWrapper.ts

```diff
@@ -0,0 +1,109 @@
+import AbstractCommand, { makeAbstractCommand } from '.'
+import { Result, WriteCommand } from '@chainlink/gauntlet-core'
+import { TerraCommand, TransactionResponse, logger } from '@chainlink/gauntlet-terra'
+import { AccAddress, LCDClient, Msg, MsgExecuteContract, MsgSend } from '@terra-money/terra.js'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { ExecutionContext } from './executionWrapper'
+import { getRDD, parseJSON } from '@chainlink/gauntlet-terra/dist/lib/rdd'
+
+const defaultBeforeExecute = async (id: string, contracts: string[], params) => {
+  logger.loading(`Executing ${id} for the following sets of inputs`)
+  logger.log('Input Params:', params)
+  await prompt(`Continue?`)
+}
+
+export const wrapCommand = (command) => {
+  return class BatchCommand extends TerraCommand {
+    static id = `${command.id}:batch`
+    subCommands: TerraCommand[]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    validateInput = (input, flags, args): Boolean => {
+      if (input == null) {
+        if (!flags.rdd) throw new Error(`One of --input, --inputFile, or --rdd must be provided`)
+        return true
+      }
+
+      const invalidInputConditions = args.length != input.length && args.length != 1
+      if (invalidInputConditions)
+        throw new Error(`Cannot apply ${input.length} command inputs to ${args.length} contracts`)
+      return true
+    }
+
+    buildCommandsFromRDD = async (flags, args): Promise<TerraCommand[]> => {
+      return await Promise.all(
+        args.map(async (contract, idx) => {
+          let c = new command(flags, [contract]) as TerraCommand
+          await c.invokeMiddlewares(c, c.middlewares)
+          c = c.buildCommand ? await c.buildCommand(flags, args) : c
+          return c
+        }),
+      )
+    }
+
+    buildCommandsFromInput = async (input, flags, args): Promise<TerraCommand[]> => {
+      return await Promise.all(
+        input.map(async (individualInput, idx) => {
+          const newFlags = { ...flags, input: individualInput }
+
+          let individualArgs = args.length == 1 ? args : [args[idx]]
+          let c = new command(newFlags, individualArgs) as TerraCommand
+          await c.invokeMiddlewares(c, c.middlewares)
+          c = c.buildCommand ? await c.buildCommand(newFlags, args) : c
+          return c
+        }),
+      )
+    }
+
+    buildCommand = async (flags, args): Promise<TerraCommand> => {
+      const input = flags.input ? flags.input : flags.inputFile ? parseJSON(flags.inputFile, 'BatchInputFile') : null
```

Comment:
Nested ternary expressions are a bit hard to read. Something like this better?
```suggestion
      let input: any[]
      if (flags.input) {
        input = flags.input
      } else if (flags.inputFile) {
        input = RDD.parseJSON(flags.inputFile, 'BatchInputFile')
      }
```

Reply:
thoughts on this?:

```
const input = flags.input
        ? flags.input
        : flags.inputFile
        ? RDD.parseJSON(flags.inputFile, 'BatchInputFile')
        : null
```